core/corecontainers/include/coretypes/listobject_factory.h

```diff
@@ -33,13 +33,21 @@ using ListPtr = ListObjectPtr<IList, T, S>;
 template <class T, class S = typename InterfaceOrTypeToSmartPtr<T>::SmartPtr, class U>
 static void addToList(ListPtr<T, S>& list, U&& element)
 {
-    list.unsafePushBack(std::forward<U>(element));
+    if constexpr (std::is_enum_v<U>)
```

Comment:
@Unsttopabull this is a temporary workaround to support enums in ListObject

Reply:
I have reverted this change, because we now use permission mask builder instead of an list object